### PR TITLE
Fix some Clang 7 warnings

### PR DESCRIPTION
**Pull request by akortunov**
Let see what will happen.

### CODE REVIEW

**Comment by akortunov** [comment]
Well, basically it works.
Of course, there is still a lot of issues with RecastNavigation, also there is a lot of warnings with possible memory leaks in the editor (but I think they are false-positives).

Now let's try to use a verbose output for scan-build.

File: extern/recastnavigation/Detour/Include/DetourAssert.h [hunk 1]
```diff
@@ -22,7 +22,15 @@
...
-#ifdef NDEBUG
+#ifdef __clang_analyzer__
+
+#	include <assert.h>
+#	define dtAssert(expression) \
+		{ \
+			assert(expression); \
+		}
+
+#elif defined NDEBUG
```
Comment (psi29a) [inline extern/recastnavigation/Detour/Include/DetourAssert.h]: Will we try to get this upstreamed if it works?

Comment (akortunov) [reply extern/recastnavigation/Detour/Include/DetourAssert.h]: Unfortunately, it does not work - there is still 23 warnings.

**Comment by akortunov** [comment]
I wonder if there is a way to take a look at generated analyzer report - in theory it should contain info about which if-branches analyzer took to generate warnings.

File: extern/recastnavigation/Detour/Include/DetourAssert.h (hunk 1 above)
Comment (psi29a) [reply extern/recastnavigation/Detour/Include/DetourAssert.h]: ok ;)

**Comment by psi29a** [comment]
If you want, I can run it locally and send you the zip file. I'll be out of town this weekend though.

**Comment by akortunov** [comment]
> If you want, I can run it locally and send you the zip file. I'll be out of town this weekend though.

Yes, please, if it is possible. I tried to run it locally (with Clang 6), but it does not detect any issues for me.
Probably I just do something wrong.

**Comment by akortunov** [comment]
Also I managed to shut up more than half of issues.
The approach used here:
1. Make Clang analyzer to use a common asserts instead of custom Recast/Detour ones.
2. Add some additional asserts to prevent lang from generating false-positives.

**Comment by psi29a** [comment]
https://mega.nz/#!jYd32KpB!Tor4HFmR2Vyq236cmLGHtEmOepQTHtNCR_i27dvgY2k

here you go, should be about 42 bugs all from master (not your branch).

**Comment by akortunov** [comment]
Ok, Detour-specific warnings are moved to separate branch. Only OpenMW-specific warnings left.

**Comment by psi29a** [comment]
Sweet, thanks! :)

**Pull request merged**